### ISSUE TITLE
Failed to validate user-specified signature

### ISSUE THREAD

**Issue by larsw** [GaloisInc/cryptol#174]
Hi,

 I've built and installed cryptol (latest from git repository master branch) under Ubuntu 14.04.

 Running cryptol from /usr/local/bin, I get:

```
  ___ _ __ _   _ _ __ | |_ ___ | |
  / __| '__| | | | '_ \| __/ _ \| |
 | (__| |  | |_| | |_) | || (_) | |
  \___|_|   \__, | .__/ \__\___/|_|
            |___/|_| version 2.1.0 (56a20cf)

Loading module Cryptol
[error] at /usr/local/share/cryptol/Cryptol.cry:17:1--17:5:
  Failed to validate user-specified signature.
    In the definition of 'Cryptol::drop', at /usr/local/share/cryptol/Cryptol.cry:17:1--17:5:
      for any type front, back, elem
        fin front
      =>
      back == front + back - front
        arising from
        matching types
        at /usr/local/share/cryptol/Cryptol.cry:17:30--17:31
      front + back >= front
        arising from
        matching types
        at /usr/local/share/cryptol/Cryptol.cry:17:7--17:22
[error] at /usr/local/share/cryptol/Cryptol.cry:14:1--14:5:
  Failed to validate user-specified signature.
    In the definition of 'Cryptol::take', at /usr/local/share/cryptol/Cryptol.cry:14:1--14:5:
      for any type front, back, elem
        fin front
      =>
      front + back >= front
        arising from
        matching types
        at /usr/local/share/cryptol/Cryptol.cry:14:11--14:12
cryptol> 
```

 Can anyone explain what I'm doing wrong?

**Comment by weaversa** [comment]
You have to also install CVC4 (http://cvc4.cs.nyu.edu/downloads/), an SMT solver that's used to solve those type constraints. There is a very short note about it in the readme file.

**Comment by weaversa** [comment]
Since typechecking the prelude is dependent on `cvc4`, I suggest printing at least a warning if it's not found in the path. The current tactic, as you see above, only produces confusion.

**Comment by acfoltzer** [comment]
Indeed this is currently how a lack of `cvc4` manifests. This should be addressed by #112

**Comment by larsw** [comment]
Ok, thanks for the commited fix - I'll try it out at work tomorrow. Strange thing is that I built & installed cvc4, but it is probably a path issue or something.
